src/backend/distributed/master/master_repair_shards.c

```diff
@@ -297,7 +296,8 @@ CopyShardPlacement(int64 shardId, char *sourceNodeName, int32 sourceNodePort,
 
 	HOLD_INTERRUPTS();
 
-	SendCommandListInSingleTransaction(targetNodeName, targetNodePort, ddlCommandList);
+	SendCommandListToWorkerInSingleTransaction(targetNodeName, targetNodePort, nodeUser,
```

Comment:
In addition, using table owner as the node user could be a problem if the owners of colocated tables are different.